tests/parser-test-files/parse-expression.txt

```diff
@@ -0,0 +1,299 @@
+================================================================================
+Parse expression
+================================================================================
+
+(package coalton-unit-tests)
+
+(define f 5)
+
+================================================================================
+Parse expression
+================================================================================
+
+(package coalton-unit-tests)
+
+(define (f _x) 5)
+
+================================================================================
+Parse expression
+================================================================================
+
+(package coalton-unit-tests)
+
+(define (f _x _y) 5)
+
+================================================================================
+Parse expression
+================================================================================
+
+(package coalton-unit-tests/prelude
+  (import coalton-prelude))
+
+(define f
+  (fn (x)
+    (+ x x)))
+
+================================================================================
+Parse expression
+================================================================================
+
+(package coalton-unit-tests/prelude
+  (import coalton-prelude))
+
+(define f
+  (fn ()
+    (+ 1 2)))
+
+================================================================================
+Parse expression (:disable)
+================================================================================
+
+(package coalton-unit-tests/prelude
+  (import coalton-prelude))
+
+(define (f)
+  (let ((x 1)
+        (declare y (T -> T))
+        (y (+ 1 2)))
+    (f x y)))
+
+================================================================================
+Parse expression
+================================================================================
+
+(package coalton-unit-tests/prelude
+  (import coalton-prelude))
+
+(define (f a b c)
+  (lisp (Integer -> Integer) (a b c)
```

Comment:
Doesn't this just return 3, a mere Integer free of arrows?
Also this doesn't and shouldn't compile
Perhaps better:
```
(define (f a b c)
  (lisp Integer (a b c)
    (cl:+ a b c))))
```
Also I don't believe that you can return an arrow type this way from a lisp form:
```
(define (f c)
  (lisp (Integer -> Integer) (c)
    (cl:lambda (a b)
      (cl:+ a b c))))
```
returns the error
```
 warning: 
    Derived type of #:G0 is
      (VALUES FUNCTION &OPTIONAL),
    conflicting with its asserted type
      (VALUES COALTON-IMPL/RUNTIME/FUNCTION-ENTRY:FUNCTION-ENTRY &OPTIONAL).
```